### PR TITLE
Add warning to complex-field nonlinearities

### PR DESCRIPTION
**Pull request by caseyflex**
Some of the nonlinearities use complex fields to realize certain effects, which may be a good model in certain circumstances but can deviate from the underlying physics. I added some warnings to make sure that the user knows what they are doing (and doesn't inadvertently use KerrNonlinearity when NonlinearSusceptibility is more appropriate, for example). These warnings will always show whenever complex-field nonlinearities are present.

### CODE REVIEW

File: tidy3d/components/medium.py [hunk 1]
```diff
@@ -442,6 +442,11 @@ def _hardcode_medium_freqs(
 
     def _validate_medium(self, medium: AbstractMedium):
         """Check that the model is compatible with the medium."""
+        log.warning(
+            "Found a medium with a 'TwoPhotonAbsorption' nonlinearity. "
+            "This uses a phenomenological model based on complex fields, "
+            "so care should be taken in interpreting the results."
```
Comment (momchil-flex) [inline tidy3d/components/medium.py]: Can we reference something for more information?

**Review by momchil-flex — APPROVED** [review]

**Review by tomflexcompute — APPROVED** [review]
Thanks Casey.

**Review by FilipeFcp — APPROVED** [review]
Nice. This will be useful for the users.

File: tidy3d/components/medium.py (hunk 1 above)
Comment (caseyflex) [reply tidy3d/components/medium.py]: done

**Pull request merged**